test/discovery.spec.js

```diff
@@ -8,16 +8,46 @@ const expect = chai.expect
 chai.use(dirtyChai)
 const multiaddr = require('multiaddr')
 const each = require('async/each')
+const series = require('async/series')
+const parallel = require('async/parallel')
+const rendezvous = require('libp2p-websocket-star-rendezvous')
 
 const WebSocketStar = require('../src')
 
 describe('peer discovery', () => {
   let listeners = []
   let ws1
-  const ma1 = multiaddr('/ip4/127.0.0.1/tcp/15001/ws/p2p-websocket-star/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSooo3A')
-
+  const ma1 = multiaddr('/ip4/127.0.0.1/tcp/16001/ws/p2p-websocket-star/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSooo4A')
   let ws2
-  const ma2 = multiaddr('/ip4/127.0.0.1/tcp/15003/ws/p2p-websocket-star/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSooo3B')
+  const ma2 = multiaddr('/ip4/127.0.0.1/tcp/16002/ws/p2p-websocket-star/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSooo4B')
+  let ws3
+  const ma3 = multiaddr('/ip4/127.0.0.1/tcp/16002/ws/p2p-websocket-star/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSooo4C')
+
+  const _r = []
+  before((done) => {
+    const base = (v) => Object.assign({
+      host: '0.0.0.0',
+      cryptoChallenge: false,
+      strictMultiaddr: false,
+      refreshPeerListIntervalMS: 1000
+    }, v)
+
+    parallel([['discovery r1', {port: 16001}], ['discovery r2', {port: 16002}]].map((v) => (cb) => {
+      rendezvous.start(base(v.pop()), (err, r) => {
+        if (err) { return cb(err) }
+        _r.push(r)
+        console.log('%s: %s', v.pop(), r.info.uri)
+        cb()
+      })
+    }), done)
+  })
```

Comment:
Browser tests are failing because it's trying to spin up a rendezvous server. Aegir spins up a few servers in .aegir.js, https://github.com/libp2p/js-libp2p-websocket-star/blob/v0.9.1/.aegir.js#L9. Using those servers should resolve the failing browser tests.

Reply:
The reason I was spinning up new rendezvous servers was that the existing servers still had some state from previous tests. So now instead of spinning up new servers I made some changes to the previous tests to clean up old connections